ManageCoursesUi.Tests/ViewModelHelperTests.cs

```diff
@@ -0,0 +1,174 @@
+
+namespace ManageCoursesUi.Tests
+{
+    [TestFixture]
+    class ViewModelHelperTests
+    {
+        [Test]
+        [TestCase("", "f", "ss", "QTS full time with salary")]
+        [TestCase("", "F", "SS", "QTS full time with salary")]
+        [TestCase(null, "f", "ss", "QTS full time with salary")]
+        [TestCase(null, "F", "SS", "QTS full time with salary")]
+        [TestCase("pg", "f", "ss", "PGCE with QTS full time with salary")]
+        [TestCase("PG", "F", "SS", "PGCE with QTS full time with salary")]
+        [TestCase("pg", "p", "ss", "PGCE with QTS part time with salary")]
+        [TestCase("PG", "P", "SS", "PGCE with QTS part time with salary")]
+        [TestCase("", "f", "sd", "QTS full time")]
+        [TestCase("", "F", "SD", "QTS full time")]
+        [TestCase("", "", "SD", "QTS part time")]
+        [TestCase("", "", "", "QTS part time")]
+        [TestCase(null, "f", "sd", "QTS full time")]
+        [TestCase(null, "F", "SD", "QTS full time")]
+        [TestCase("pg", "f", "sd", "PGCE with QTS full time")]
+        [TestCase("PG", "F", "SD", "PGCE with QTS full time")]
+        [TestCase("pg", "p", "sd", "PGCE with QTS part time")]
+        [TestCase("PG", "P", "SD", "PGCE with QTS part time")]        
+        public void TestGetCourseVariantType(string profpostFlag, string studyMode, string programType, string expectedResult)
+        {
+            var course = new Course
+            {
+                ProfpostFlag = profpostFlag,
+                StudyMode = studyMode,
+                ProgramType = programType
+            };
+            var result = course.GetCourseVariantType();
+            result.Should().Be(expectedResult);
+        }
+
+        [Test]
+        [TestCase("N","N","N","N", "New-not yet running")]
+        [TestCase("N", "N", "S", "S", "New-not yet running")]
+        [TestCase("N", "S", "S", "S", "New-not yet running")]
+        [TestCase("S", "N", "S", "S", "New-not yet running")]
+        [TestCase("S", "S", "N", "S", "New-not yet running")]
+        [TestCase("S", "S", "S", "N", "New-not yet running")]
+        [TestCase("n", "n", "n", "n", "New-not yet running")]
+        [TestCase("n", "n", "n", "s", "New-not yet running")]
+        [TestCase("n", "s", "s", "s", "New-not yet running")]
+        [TestCase("s", "n", "s", "s", "New-not yet running")]
+        [TestCase("s", "s", "n", "s", "New-not yet running")]
+        [TestCase("s", "s", "s", "n", "New-not yet running")]
+        [TestCase("N", "", "", "N", "New-not yet running")]
+        [TestCase("R", "N", "N", "N", "Running")]
+        [TestCase("N", "R", "N", "N", "Running")]
+        [TestCase("N", "N", "R", "N", "Running")]
+        [TestCase("N", "N", "N", "R", "Running")]
+        [TestCase("r", "n", "n", "n", "Running")]
+        [TestCase("n", "r", "n", "n", "Running")]
+        [TestCase("n", "n", "r", "n", "Running")]
+        [TestCase("n", "n", "n", "r", "Running")]
+        [TestCase("R", "", "", "N", "Running")]
```

Comment:
I think we're missing a couple of test cases to clarify that Running supercedes any other state:

`[TestCase("R", "S", "D", "D", "Running")]`
`[TestCase("R", "S", "D", "N", "Running")]`

Reply:
Yes makes sense